Framework/PythonInterface/plugins/algorithms/RefineSatellitePeaks.py

```diff
@@ -0,0 +1,180 @@
+from mantid.kernel import *
+from mantid.dataobjects import PeaksWorkspaceProperty
+from mantid.api import *
+from mantid.simpleapi import *
+
+import numpy as np
+from scipy.spatial import KDTree
+from scipy.cluster.vq import kmeans2
+import scipy.cluster.hierarchy as hcluster
+
+
+class RefineSatellitePeaks(DataProcessorAlgorithm):
+
+    def category(self):
+        return 'Crystal\\Peaks'
+
+    def PyInit(self):
+        self.declareProperty(PeaksWorkspaceProperty(name="NuclearPeaks",
+                                                    defaultValue="",
+                                                    direction=Direction.Input),
+                             doc="Main integer HKL peaks. Q vectors will be calculated relative to these peaks.")
+
+        self.declareProperty(PeaksWorkspaceProperty(name="SatellitePeaks",
+                                                    defaultValue="",
+                                                    direction=Direction.Input),
+                             doc="Positions of seed satellite peaks. These will be used to define the q vectors for each satellite.")
+
+        self.declareProperty(WorkspaceProperty(name="MDWorkspace",
+                                               defaultValue="",
+                                               direction=Direction.Input,
+                                               validator=MDFrameValidator("HKL")),
+                             doc="MD workspace to search for satellites peak in. This data must be in the HKL frame.")
+
+        self.declareProperty(PeaksWorkspaceProperty(name="OutputWorkspace",
+                                                    defaultValue="",
+                                                    direction=Direction.Output),
+                             doc="All found satellite peaks. These will be given with satellite coordinates.")
+
+        self.declareProperty('NumOfQs', -1, direction=Direction.Input,
+                             doc="The number of satellite peaks to look for. If this option is not set to the default then all the \
```

Comment:
Is `q vectors` adequate enough as an explanation to algorithm users?

Reply:
Perhaps "modulation vectors" is more descriptive, although `q` is common domain specific terminology in superspace crystallography.